src/core/crypto/context_size.hpp

```diff
@@ -0,0 +1,144 @@
+/*
+ *  Copyright (c) 2021, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef CRYPTO_CONTEXT_H_
+#define CRYPTO_CONTEXT_H_
+
+#include "openthread-core-config.h"
+
+#if OPENTHREAD_CONFIG_CRYPTO_LIB == OPENTHREAD_CONFIG_CRYPTO_LIB_MBEDTLS_2
```

Comment:
I would suggest moving all these config definition into the `core/config/crypto.h` header file (or maybe another header file but under `core/config` folder). 

I think we want to have all `OPENTHREAD_CONFIG_*` definitions to be under the `core/config/` folder.

Reply:
Here is maybe a better idea:
- We can move all `OPENTHREAD_CONFIG_AES_CONTEXT_SIZE`, ... definition  in `config/crypto.h` 
- But add documentation that these are only used (and required) when `OPENTHREAD_CONFIG_CRYPTO_LIB` is set to `OPENTHREAD_CONFIG_CRYPTO_LIB_PLATFORM`
- Then we can still have `context_size.hpp` here and define a bunch of `constexpr` constants (under `Crypto` namespace):
```c++
#if OPENTHREAD_CONFIG_CRYPTO_LIB_PLATFORM == OPENTHREAD_CONFIG_CRYPTO_LIB_MBEDTLS_2

#include <mbedtls/aes>
#include <mbedtls/md.h>
#include <mbedtls/sha256.h>

namespace Crypto {

constexpr uint16_t kAesContextSize = sizeof(mbedtls_aes_context);
constexpr uint16_t kHmacShaContextSize = sizeof(mbedtls_md_context_t));
...
```

I think this maybe even better and simpler. 
Thoughts?

Reply:
Limiting the size configurations to `OPENTHREAD_CONFIG_CRYPTO_LIB_PLATFORM` makes sense.